toplevel/opttopmain.ml

```diff
@@ -241,6 +242,16 @@ module Options = Main_args.Make_opttop_options (struct
   let _safe_string = clear unsafe_string
   let _unsafe_string = set unsafe_string
   let _open s = open_modules := s :: !open_modules
+  let _color option =
+    begin match color_reader.parse option with
+          | None -> ()
+          | Some setting -> color := Some setting
+    end
```

Comment:
I count 12 occurrences of this code pattern in this patch. Could we factorize this into a Misc function? There is a risk of confusion with the properly-error-reporting Compmisc function, so a longer name should be used for the present one to pay for the dubious ignore semantics.  What about `set_or_ignore : (string -> 'a option) -> 'a option env -> unit`? (Being careful to specify that on `None`, it doesn't set the reference, instead of setting it to None as one could naively expect.)

Reply:
Done, I think?

Reply:
Seems fine to me, thanks!